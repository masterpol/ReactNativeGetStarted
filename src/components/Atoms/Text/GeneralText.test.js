import React from 'react'
import { P, H1, H2, H3, H4, H5, H6 } from '.'
import Theme from '_themes'

describe('General Text', () => {
  let props

  beforeEach(() => {
    props = {
      children: 'test',
    }
  })

  describe('structure', () => {
    it('should match the snapshot for general text', () => {
      const wrapper = render(
        <Theme>
          <P testID="text-test" {...props} />
        </Theme>,
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot for H1', () => {
      const wrapper = render(
        <Theme>
          <H1 testID="h1-test" {...props} />
        </Theme>,
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot for H2', () => {
      const wrapper = render(
        <Theme>
          <H2 testID="h2-test" {...props} />
        </Theme>,
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot for H3', () => {
      const wrapper = render(
        <Theme>
          <H3 testID="h3-test" {...props} />
        </Theme>,
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot for H4', () => {
      const wrapper = render(
        <Theme>
          <H4 testID="h4-test" {...props} />
        </Theme>,
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot for H5', () => {
      const wrapper = render(
        <Theme>
          <H5 testID="h5-test" {...props} />
        </Theme>,
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot for H6', () => {
      const wrapper = render(
        <Theme>
          <H6 testID="h6-test" {...props} />
        </Theme>,
      )
      expect(wrapper).toMatchSnapshot()
    })
  })
})
