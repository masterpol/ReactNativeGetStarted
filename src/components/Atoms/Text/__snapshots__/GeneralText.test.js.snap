// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button structure should match the snapshot for H1 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
      },
      Array [
        Object {
          "fontSize": 42,
        },
        undefined,
      ],
    ]
  }
  testID="h1-test"
>
  test
</Text>
`;

exports[`Button structure should match the snapshot for H2 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
      },
      Array [
        Object {
          "fontSize": 38,
        },
        undefined,
      ],
    ]
  }
  testID="h2-test"
>
  test
</Text>
`;

exports[`Button structure should match the snapshot for H3 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
      },
      Array [
        Object {
          "fontSize": 32,
        },
        undefined,
      ],
    ]
  }
  testID="h3-test"
>
  test
</Text>
`;

exports[`Button structure should match the snapshot for H4 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
      },
      Array [
        Object {
          "fontSize": 28,
        },
        undefined,
      ],
    ]
  }
  testID="h4-test"
>
  test
</Text>
`;

exports[`Button structure should match the snapshot for H5 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
      },
      Array [
        Object {
          "fontSize": 24,
        },
        undefined,
      ],
    ]
  }
  testID="h5-test"
>
  test
</Text>
`;

exports[`Button structure should match the snapshot for H6 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
      },
      Array [
        Object {
          "fontSize": 20,
        },
        undefined,
      ],
    ]
  }
  testID="h6-test"
>
  test
</Text>
`;

exports[`Button structure should match the snapshot for general text 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#333333",
      },
      undefined,
    ]
  }
  testID="text-test"
>
  test
</Text>
`;
